console/views.py

```diff
@@ -116,70 +130,48 @@ def get_filtered_responses(tx_id, ru_ref, survey_id, datetime_earliest, datetime
-    private_key_password = settings.PRIVATE_KEY_PASSWORD
-    encrypter = Encrypter(eq_private_key, eq_private_key_password, private_key, private_key_password)
-    encrypted_data = encrypter.encrypt(unencrypted_json)
-    logger.info('Data successfully encrypted')
-
-    return encrypted_data
-
-
-def get_publisher(logger):
-    urls = settings.RABBIT_URLS
-    queue = settings.RABBIT_SURVEY_QUEUE
-    collect_publisher = QueuePublisher(logger, urls, queue)
-
-    return collect_publisher
+def reprocess_transaction(logger, json_data):
+    logger.info("Reprocessing transaction", tx_id=json_data["tx_id"])
+    if json_data.get("invalid"):
+        del json_data["invalid"]
+    validate_response = send_data(logger=logger, url=settings.SDX_VALIDATE_URL, json=json_data, request_type="POST")
+    if validate_response != 200:
+        json_data['invalid'] = True
+    send_data(logger=logger, url=settings.SDX_STORE_URL, json=json_data, request_type="POST")
 
 
-def publish_result(publisher, json_string):
-    tx_id = str(uuid.uuid4())
-    logger.info('Created new tx_id ' + tx_id)
-    json_string['tx_id'] = tx_id
-    json_string['survey_id'] = str(json_string['survey_id'])
-    encrypted_data = encrypt_data(json_string)
-
-    publisher.publish_message(encrypted_data, headers={'tx_id': tx_id})
-
-
-@app.route('/store', methods=['GET', 'POST'])
+@app.route('/store/', defaults={'page': 0}, methods=['GET', 'POST'])
+@app.route('/store/<page>', methods=['GET', 'POST'])
 @flask_security.roles_required('SDX-Developer')
-def store():
+def store(page):
+    audited_logger = logger.bind(user=flask_security.core.current_user.email)
     if request.method == 'POST':
-        json_string = request.form.get('json_data')
-        if json_string == '':
-            return redirect(url_for('store'))
-        unencrypted_json = json.loads(json_string.replace("'", '"'))
-
-        collect_publisher = get_publisher(logger)
-        collect_publisher._connect()
-
-        if isinstance(unencrypted_json, list):
-            logger.info('Reprocessing all results')
-            for string in unencrypted_json:
-                logger.info('Reprocessing transaction', tx_id=string["tx_id"])
-                publish_result(collect_publisher, string)
+        json_survey_data = request.form.get('json_data')
+        if not json_survey_data:
+            json_survey_data = request.form.get('json_data_list')
+            if not json_survey_data:
+                return url_for('store')
+            json_survey_data = eval(json_survey_data)
+
+        if isinstance(json_survey_data, list):
+            audited_logger.info("Reprocessing multiple transactions")
+            for json_data in json_survey_data:
+                reprocess_transaction(audited_logger, json_data)
         else:
-            publish_result(collect_publisher, unencrypted_json)
-
-        collect_publisher._disconnect()
-
+            json_data = json.loads(json_survey_data.replace("'", '"'))
```

Comment:
when the reprocess_transaction is called you get a log like that with the tx_id in it so shouldnt be neccessary here